icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6156,6 +6167,44 @@ static void TestInvalidLanguageTag(void) {
     }
 }
 
+static const struct {
+    const char  *input;
+    const char  *canonical;
+} langtag_to_canonical[] = {
+    {"de-DD", "de-DE"},
+    {"de-DD-u-co-phonebk", "de-DE-u-co-phonebk"},
+    {"jw-id", "jv-ID"},
+    {"jw-id-u-ca-islamic-civil", "jv-ID-u-ca-islamic-civil"},
+    {"mo-md", "ro-MD"},
+    {"my-bu-u-nu-mymr", "my-MM-u-nu-mymr"},
+    {"yuu-ru", "yug-RU"},
+    {NULL, NULL},
+};
+
+
+static void TestLangAndRegionCanonicalize(void) {
+    char locale[256];
+    char canonical[256];
+    int32_t i;
+    UErrorCode status;
+    for (i = 0; langtag_to_canonical[i].input != NULL; i++) {
```

Comment:
Also here, iterating over the allocated size is both simpler and safer.

Reply:
Done